R/configuration-plan.R

```diff
@@ -0,0 +1,236 @@
+      return(sectionTitle)
+    },
+    
+    #' @description Get location of .md file corresponding to a specific section Id
+    #' @param id section identifier
+    #' @return The markdown file corresponding to the id in the configuration plan field `sections`
+    getSectionMarkdown = function(id) {
+      validateIsIncluded(id, private$.sections$id, groupName = "'id' variable of sections")
+      selectedId <- private$.sections$id %in% id
+      return(file.path(self$workflowFolder, private$.sections$md[selectedId]))
+    },
+    
+    #' @description Get location of .md intro file
+    #' @return The markdown file corresponding to introduction
+    getIntroMarkdown = function() {
+      return(file.path(self$workflowFolder, self$markdownIntro))
+    },
+    
+    #' @description Copy content to section markdown
+    #' @param id section identifier
+    #' @param logFolder path where logs are saved
+    copySectionContent = function(id, logFolder = getwd()) {
+      validateIsIncluded(id, private$.sections$id, groupName = "'id' variable of sections")
+      selectedId <- private$.sections$id %in% id
+      sectionContent <- private$.sections$content[selectedId]
+      # Case in which no content was defined
+      if(is.na(sectionContent)){return(invisible())}
+      # Get location of content
+      markdownLocation <- file.path(self$referenceFolder, sectionContent)
+      # In case file does not exist
+      if(!file.exists(markdownLocation)){
+        logWorkflow(message = paste0("Section content '", sectionContent,"' not found"),
+                    pathFolder = logFolder,
+                    logTypes = c(LogTypes$Error, LogTypes$Debug))
+        return(invisible())
+      }
+      addTextChunk(fileName = self$getSectionMarkdown(id), text = readLines(markdownLocation), logFolder = logFolder)
+      return(invisible())
+    },
+    
+    #' @description Copy input to section markdown
+    #' @param input list including SectionId and Path
+    #' @param logFolder path where logs are saved
+    copyInput = function(input, logFolder = getwd()) {
+      validateIsIncluded(names(input), c("SectionId", "Path"))
+      # Get location of input
+      inputLocation <- file.path(self$referenceFolder, input$Path)
+      # In case file does not exist
+      if(!file.exists(inputLocation)){
+        logWorkflow(message = paste0("Input '", input$Path,"' not found"),
+                    pathFolder = logFolder,
+                    logTypes = c(LogTypes$Error, LogTypes$Debug))
+        return(invisible())
+      }
+      addTextChunk(fileName = self$getSectionMarkdown(input$SectionId), text = readLines(inputLocation), logFolder = logFolder)
+      return(invisible())
+    },
+    
+    #' @description Copy intro to intro markdown
+    #' @param intro list including Path
+    #' @param logFolder path where logs are saved
+    copyIntro = function(intro, logFolder = getwd()) {
+      validateIsIncluded(names(intro), "Path")

```

Comment:
This is partly done here: 
- the field "Path" defined within "Intro" of the configuration plan (json file) is mandatory as it is where the content of intro is read
- The expected type of intro is a "list" as it is read from `jsonlite::fromJSON`, but it should include the mandatory field "Path". Any R object with such a field is actually also okay (e.g. R6 object or a data.frame). So, in case someone needs to tune the `intro` field of the configuration plan, I did not check the exact type of `intro`.

Let me know if that is suitable